x/group/spec/state.md

```diff
@@ -0,0 +1,150 @@
+# State & Types
+
+## Groups
+
+Groups are simply aggregations of members.
+
+### ID
+
+They have an auto-incremented ID:
+```go
+type GroupID uint64
+```
+
+### Properties
+
+#### `Members []Member`
+
+An array of `Member` structs:
+
+```go
+// Member specifies a address and a weight for a group member
+type Member struct {
+	// The address of a group member. Can be another group or a contract
+	Address sdk.AccAddress `json:"address"`
+	// The integral weight of this member with respect to other members and the decision threshold
+	Weight sdk.Int `json:"weight"`
+}
+```
+
+#### `Owner sdk.ACcAddress`
+
+Owner is the account which "owns" the group and has the permission to 
+add and remove memers.
+
+#### `Memo string`
+
+A single string memo
+
+#### Indexes
+
+#### `Member`
+
+It should be possible to look-up groups by member
+
+#### `Owner`
+
```

Comment:
@ethanfrey can you let me know if this general structure of specifying state makes sense? I'm imagining this is similar to how you would structure an ORM - mapping properties, indexes, etc. rather than structs so to me its most logical to express it this way instead of go structs. I'll try to add more details on each property etc. tomorrow.